### PR TITLE
Use plt.rc_context for default styles

### LINKED ISSUE #7313: Using scatter with unfilled marker raises matplotlib UserWarning
### What is your issue?

If plotting a scatter plot with an unfilled marker, matplotlib raises a UserWarning. Consider this MWE:

```python
import warnings

import numpy as np
import xarray as xr

warnings.filterwarnings("error")

ds = xr.Dataset({"a": ("dim", np.arange(3, 10))}, {"dim": np.arange(7)})
ds.plot.scatter(x="dim", y="a", marker="x")
```

Running this with matplotlib 3.6.2 yields:

```
Traceback (most recent call last):
  File "/my/path/test.py", line 9, in <module>
    ds.plot.scatter(x="dim", y="a", marker="x")
  File "/my/path/.venv/lib/python3.9/site-packages/xarray/plot/accessor.py", line 1071, in scatter
    return dataset_plot.scatter(self._ds, *args, **kwargs)
  File "/my/path/.venv/lib/python3.9/site-packages/xarray/plot/dataset_plot.py", line 916, in scatter
    return da.plot.scatter(*locals_.pop("args", ()), **locals_)
  File "/my/path/.venv/lib/python3.9/site-packages/xarray/plot/accessor.py", line 305, in scatter
    return dataarray_plot.scatter(self._da, *args, **kwargs)
  File "/my/path/.venv/lib/python3.9/site-packages/xarray/plot/dataarray_plot.py", line 1002, in newplotfunc
    primitive = plotfunc(
  File "/my/path/.venv/lib/python3.9/site-packages/xarray/plot/dataarray_plot.py", line 1277, in scatter
    primitive = ax.scatter(*[p.to_numpy().ravel() for p in plts], **kwargs)
  File "/my/path/.venv/lib/python3.9/site-packages/matplotlib/__init__.py", line 1423, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/my/path/.venv/lib/python3.9/site-packages/matplotlib/axes/_axes.py", line 4571, in scatter
    _api.warn_external(
  File "/my/path/.venv/lib/python3.9/site-packages/matplotlib/_api/__init__.py", line 363, in warn_external
    warnings.warn(message, category, stacklevel)
UserWarning: You passed a edgecolor/edgecolors ('w') for an unfilled marker ('x').  Matplotlib is ignoring the edgecolor in favor of the facecolor.  This behavior may change in the future.
```

Environment:
```
INSTALLED VERSIONS
------------------
commit: None
python: 3.9.10 (main, Mar 15 2022, 15:56:56) 
[GCC 7.5.0]
python-bits: 64
OS: Linux
OS-release: 3.10.0-1160.49.1.el7.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: ('en_US', 'UTF-8')
libhdf5: 1.12.2
libnetcdf: 4.9.0

xarray: 2022.11.0
pandas: 1.5.1
numpy: 1.23.5
scipy: 1.9.3
netCDF4: 1.6.2
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: 1.6.2
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: None
distributed: None
matplotlib: 3.6.2
cartopy: None
seaborn: None
numbagg: None
fsspec: None
cupy: None
pint: None
sparse: None
flox: None
numpy_groupies: None
setuptools: 65.6.0
pip: 22.3.1
conda: None
pytest: 7.2.0
IPython: 8.6.0
sphinx: None

```

### PR DESCRIPTION
**Pull request by Illviljan**
- [x] Closes #7313
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by Illviljan** [comment]
Seaborn did something similar as the original but had a little check if the marker was filled or not:
https://github.com/mwaskom/seaborn/blob/bf4695466d742f301f361b8d0c8168c5c4bdf889/seaborn/relational.py#L563-L570

**Review by headtr1ck — APPROVED** [review]
Did not check what seaborn is doing exactly, but it seems like a reasonable approach.

File: xarray/tests/test_plot.py [hunk 1]
```diff
@@ -3220,3 +3221,39 @@ def test_facetgrid_axes_raises_deprecation_warning() -> None:
...
+        # Facetgrids should have the default value as well:
+        fg = ds.plot.scatter(x="A", y="B", col="x", marker="o")
+        ax = fg.axs.ravel()[0]
+        np.testing.assert_allclose(
+            ax.collections[0].get_edgecolor(), mpl.colors.to_rgba_array("w")
+        )
+
+        # scatter should not emit any warnings when using unfilled markers:
+        with warnings.catch_warnings():
+            warnings.simplefilter("error")
```
Comment (mathause) [inline xarray/tests/test_plot.py:3250]: You could use

https://github.com/pydata/xarray/blob/3aa75c8d00a4a2d4acf10d80f76b937cadb666b7/xarray/tests/__init__.py#L180


```suggestion
        with assert_no_warnings():
```

(would still need the import)

File: xarray/plot/dataarray_plot.py [hunk 2]
```diff
@@ -984,29 +992,29 @@ def newplotfunc(
...
-        if z is not None:
-            if ax is None:
-                subplot_kws.update(projection="3d")
-            ax = get_axis(figsize, size, aspect, ax, **subplot_kws)
-            # Using 30, 30 minimizes rotation of the plot. Making it easier to
-            # build on your intuition from 2D plots:
-            plt = import_matplotlib_pyplot()
-            if Version(plt.matplotlib.__version__) < Version("3.5.0"):
-                ax.view_init(azim=30, elev=30)
+        with plt.rc_context(_styles):
```
Comment (dcherian) [inline xarray/plot/dataarray_plot.py:995]: Doesn't this override any user style settings in `.matplotlibrc`?

Comment (Illviljan) [reply xarray/plot/dataarray_plot.py:995]: Yes, the innermost context takes priority. `edgecolor="k"` can be used though. 
We have the same problem with color maps:

```python
ds = xr.tutorial.scatter_example_dataset(seed=42)

fig, axs = plt.subplots(1, 2)
with plt.rc_context(
    {
        "scatter.edgecolors": "k",
        "image.cmap": "Greys",
    }
):
    # Normal scatter:
    axs[0].scatter(ds.A.values.ravel(), ds.B.values.ravel(), c=3 * ds.B.values)

    # xarray scatter:
    ds.plot.scatter(x="A", y="B", ax=axs[1])
```
image

**Comment by Illviljan** [comment]
This has stalled for long enough. I'll merge this at the end of next week unless someone disagrees, at least it is an improvement to the bugged main.

**Pull request merged**